File: stack.c
#include<stdio.h>
#define MAX_STACK_SIZE 100
 
int stack[MAX_STACK_SIZE];      //스택의 사이즈 100으로 지정 
int top=-1;     
 
int IsEmpty(){      
    if(top<0)       
        return true;
    else            
        return false;
    }


int IsFull(){       //스택이 꽉 차있나? 확인하는 함수임
    if(top>=MAX_STACK_SIZE-1)   
        return true;         
    else                
        return false;
}
 
void push(int value){       
    if(IsFull()==true)      
        printf("스택이 가득 찼습니다.");
    else                   
        stack[++top]=value; 
}
 
int pop(){                  //데이터 빼기
    if(IsEmpty()==true)     
        printf("스택이 비었습니다.");
    else                
        return stack[top--];  
}
 
int main(){
    
    push(3);
    push(5);
    push(12);
    printf("%d ",pop());
    printf("%d ",pop());
    printf("%d ",pop());
 
    return 0;
}

File: stack.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedListStack():
    def __init__(self):   #초기화
        self.head = None
        
    def push(self, data): #data push
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node
        
    def pop(self):        #data pop
        pop_object = None
        if self.isEmpty():
            print("Stack is Empty")
        else:
            pop_object = self.head.data
            self.head = self.head.next
        return pop_object
    
    def top(self):    #stack's top
        top_object = None
        if self.isEmpty():
            print("Stack is Empty")
        else:
            top_object = self.head.data            
        return top_object
    
    def isEmpty(self):    #check isempty
        is_empty = False
        if self.head is None:
            is_empty = True
        return is_empty
